### ISSUE TITLE
ModalBottomSheet steals scroll gestures

### ISSUE THREAD

**Issue by Perfomer** [composablehorizons/compose-unstyled#35]
Hello. Thank you for your library!

**Problem:**
Unfortunately I've faced an issue that blocks me using the current library:
`ModalBottomSheet` steals scroll gestures if it has scrollable content inside.

**Expected behavior:**
- Each scroll gesture scrolls inner scroll container (e.g. `LazyVerticalGrid`) until you reach the top of the container. Then `ModalBottomSheet` handles scroll gestures and can be dismissed by scroll gesture.

**Actual behavior:**
- About one in three scroll gesture scrolls out of inner scroll container (e.g. `LazyVerticalGrid`) and `ModalBottomSheet` handles it by itself what leads to `ModalBottomSheet` dismissing.

Could you please take a look?

I found out that commit 0ed78491 have introduced the bug. 
Also I attach two videos before and after that commit where you can see the reproduction:
- Before commit (everything is fine): [okay.webm](https://github.com/user-attachments/assets/b919e2a3-b340-4b73-b60f-31cc7c08cb1b)
- After commit (bug): [broken.webm](https://github.com/user-attachments/assets/bac8a220-d29b-44a2-9272-2fe28c4facf4)

**Preconditions:**
1. Android device or emulator (but it doesn't look as device specific). I tried with:
   - [Physical] Pixel 8, Android 15 
   - [Emulator] Pixel 6, Android 12
3. Version of `com.composables:core`: 1.17.1 or higher.
4. Have a `ModalBottomSheet` with scrollable content inside of `Sheet` (e.g. `LazyVerticalGrid`).

**Steps to reproduce:**
1. Scroll down to receive a space for scrolling in the opposite way.
2. Then scroll up few times (it's easier to reproduce when you make slight and short swipes).

**Patch:**
I've prepared a patch with reproduction for your `:demo-modalsheet` android app:
- [Scroll_gesture_stealing_reproduce.patch](https://github.com/user-attachments/files/17495208/Scroll_gesture_stealing_reproduce.patch)

**Comment by alexstyl** [comment]
thanks for openning this issue. I was able to reproduce this.

weird as indeed it seems like it is caused by the compose version update. will look into this asap

**Comment by alexstyl** [comment]
I was able to fix it locally by using the framework's `AnchoredDraggable` instead of `CoreAnchoredDraggable` (which was copy pasted from an older version of the framework).

Haven't pushed any changes yet as I need to update the version in the repo first, but it is fixable for sure.

**Comment by alexstyl** [comment]
Fixed in [1.18.0](https://github.com/composablehorizons/compose-unstyled/releases/tag/1.18.0). Should be available via Maven Central shortly

**Comment by Perfomer** [comment]
Hello! I checked fix in your demo-app and it works great! 

But after upgrade to `1.18.0` in my project I receive build error `app:checkDebugDuplicateClasses`.

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:checkDebugDuplicateClasses'.
Caused by: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing 
Caused by: java.lang.RuntimeException: Duplicate class androidx.annotation.FloatRange found in modules annotation-jvm-1.8.1 (androidx.annotation:annotation-jvm:1.8.1) and core-debug-runtime (com.composables:core-android-debug:1.18.0)
Duplicate class androidx.annotation.IntRange found in modules annotation-jvm-1.8.1 (androidx.annotation:annotation-jvm:1.8.1) and core-debug-runtime (com.composables:core-android-debug:1.18.0)
Duplicate class androidx.collection.FloatSetKt found in modules collection-jvm-1.4 (androidx.collection:collection-jvm:1.4.4) and core-debug-runtime (com.composables:core-android-debug:1.18.0)
Duplicate class androidx.collection.MapEntry found in modules collection-jvm-1.4 (androidx.collection:collection-jvm:1.4.4) and core-debug-runtime (com.composables:core-android-debug:1.18.0)
Duplicate class androidx.collection.MutableMapEntry found in modules collection-jvm-1.4 (androidx.collection:collection-jvm:1.4.4) and core-debug-runtime (com.composables:core-android-debug:1.18.0)
Duplicate class androidx.compose.foundation.gestures.DraggableNode found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
Duplicate class androidx.compose.foundation.gestures.DraggableNode$drag$2 found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
Duplicate class androidx.compose.foundation.gestures.DraggableNode$onDragStarted$1 found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
Duplicate class androidx.compose.foundation.gestures.DraggableNode$onDragStopped$1 found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
Duplicate class androidx.compose.foundation.gestures.DraggableState found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
Duplicate class androidx.compose.foundation.gestures.DraggableState$DefaultImpls found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
Duplicate class androidx.compose.foundation.gestures.MapDraggableAnchors found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
Duplicate class androidx.compose.foundation.gestures.TouchSlopDetector found in modules core-debug-runtime (com.composables:core-android-debug:1.18.0) and foundation-release-runtime (androidx.compose.foundation:foundation-android:1.7.4)
```

I shortened it a bit. Full version of error log: 
- [error log.txt](https://github.com/user-attachments/files/17530729/error.log.txt)

Version of projects Compose BOM: `androidx.compose:compose-bom:2024.10.00`.

Why can this happen?

**Comment by Perfomer** [comment]
Looks like it doesn't depend on compose version in the project.

I created new empty project even without compose. And after adding `implementation("com.composables:core:1.18.0")` I receive build error with duplicated classes.

**Comment by alexstyl** [comment]
My bad. Fixing it right away

**Comment by alexstyl** [comment]
Fixed in [1.18.1](https://github.com/composablehorizons/compose-unstyled/releases/tag/1.18.1). Might take a while for the release to be available in Maven Central though

**Comment by Perfomer** [comment]
Thank you, now it works!
